lib/marc_match_key/format_character_key.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+### A collection of functions and classes to generate a match key based on the
+###   GoldRush algorithm
+module MarcMatchKey
+  ### Generates the format character portion of a GoldRush key
+  class FormatCharacterKey
+    attr_reader :record, :key
+
+    def initialize(record)
+      @record = record
+      @key ||= generate_key
+    end
+
+    private
+
+    def generate_key
+      if (record['245'] && record['245']['h'] =~ /electronic resource/) ||
+         electronic_resource_3xx? ||
+         electronic_resource_5xx_f007? ||
+         (record['086'] && record['856'])
+        'e'
+      else
+        'p'
+      end
+    end
+
+    def electronic_resource_5xx_f007?
+      record.fields(%w[533 590]).any? { |f| f['a'] =~ /[Ee]lectronic reproduction/ } ||
```

Comment:
Nice!  I like the use of `#any?` and also the handling of both Electronic and electronic.